### PR TITLE
feat: add agent team templates for multi-agent orchestration

### PR DESCRIPTION
**Pull request by CodeWithBehnam**
## Summary

- Adds 9 specialized agents organized into **3 pre-configured teams** for multi-agent collaboration
- Uses Claude Code's experimental agent teams feature (`CLAUDE_CODE_EXPERIMENTAL_AGENT_TEAMS=1`)
- New `agent-teams/` category under agents with README explaining all three teams

### Code Review Team (3 agents, parallel)
- **security-reviewer**: OWASP top 10, injection, auth bypass, data exposure
- **performance-reviewer**: N+1 queries, algorithmic complexity, memory leaks, missing caching
- **test-coverage-analyst**: coverage gaps, edge cases, assertion quality, skeleton test code

### Feature Development Team (architect first, then 2 in parallel)
- **architect**: explores codebase patterns, designs file structure and data flow
- **implementer**: writes production code following the architect's plan
- **test-writer**: writes unit, integration, and regression tests

### Debugging Team (3 competing hypotheses, parallel)
- **log-analyst**: works backwards from error symptoms, timeline reconstruction
- **code-tracer**: traces execution path from entry point to failure
- **hypothesis-tester**: designs and runs experiments to confirm/eliminate theories

## Test plan

- [ ] Verify agent markdown files have valid YAML frontmatter (name, description, tools, model, color)
- [ ] Copy agents to `.claude/agents/` and test with agent teams enabled
- [ ] Verify README usage instructions are accurate

Source: [cc-docs](https://github.com/CodeWithBehnam/cc-docs)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Adds pre-configured multi-agent team templates (9 agents across review, development, and debugging) for orchestrating complex tasks. Requires env var `CLAUDE_CODE_EXPERIMENTAL_AGENT_TEAMS=1` (no new secrets), and improves review instructions to include staged and untracked files.

- **New Features**
  - Area: components (`cli-tool/components/`) — new `agents/agent-teams/` with README and 9 agent definitions
  - New components added — regenerate catalog (`docs/components.json`) after merge

- **Bug Fixes**
  - Review agents now use `git diff HEAD` and `git ls-files --others --exclude-standard` to include untracked files
  - `hypothesis-tester`: clarified it works from bug reports directly

<sup>Written for commit ef4c1caf2336a9dedcebd1d25036f02a826e2a02. Summary will update on new commits.</sup>

### CODE REVIEW

File: cli-tool/components/agents/agent-teams/test-coverage-analyst.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  Use proactively when reviewing code to assess test quality.
+tools: Read, Grep, Glob, Bash
+model: sonnet
+color: green
+---
+
+You are a senior QA engineer analyzing test coverage for code changes.
+
+When invoked:
+1. Run `git diff` to see the changes
```
[the remark this answers is not in the document]
Comment (CodeWithBehnam) [reply cli-tool/components/agents/agent-teams/test-coverage-analyst.md]: Fixed in 9841119. Changed to `git diff HEAD` which captures both staged and unstaged changes, giving a complete view of what's being reviewed.

File: cli-tool/components/agents/agent-teams/security-reviewer.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  Use proactively when reviewing PRs or code changes.
+tools: Read, Grep, Glob, Bash
+model: sonnet
+color: red
+---
+
+You are a senior application security engineer reviewing code changes.
+
+When invoked:
+1. Run `git diff` to see the changes being reviewed
```
[the remark this answers is not in the document]
Comment (CodeWithBehnam) [reply cli-tool/components/agents/agent-teams/security-reviewer.md]: Fixed in 9841119. Changed to `git diff HEAD` to capture all staged and unstaged changes.

File: cli-tool/components/agents/agent-teams/hypothesis-tester.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  Use proactively when debugging to confirm or eliminate hypotheses.
+tools: Read, Edit, Write, Bash, Grep, Glob
+model: sonnet
+color: magenta
+---
+
+You are a debugging specialist who tests hypotheses through experiments.
+
+When invoked:
+1. Review the hypotheses from other debugging agents
```
[the remark this answers is not in the document]
Comment (CodeWithBehnam) [reply cli-tool/components/agents/agent-teams/hypothesis-tester.md]: Fixed in 9841119. Reworded to clarify the hypothesis-tester works from bug reports and provided context, not from other agents' output. Each agent in the team operates independently with its own context.

File: cli-tool/components/agents/agent-teams/security-reviewer.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  Use proactively when reviewing PRs or code changes.
+tools: Read, Grep, Glob, Bash
+model: sonnet
+color: red
+---
+
+You are a senior application security engineer reviewing code changes.
+
+When invoked:
+1. Run `git diff HEAD` to see all staged and unstaged changes being reviewed
```
[the remark this answers is not in the document]
Comment (CodeWithBehnam) [reply cli-tool/components/agents/agent-teams/security-reviewer.md]: Fixed in ef4c1ca. Now also runs `git ls-files --others --exclude-standard` to discover untracked new files that `git diff HEAD` misses.

File: cli-tool/components/agents/agent-teams/test-coverage-analyst.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  Use proactively when reviewing code to assess test quality.
+tools: Read, Grep, Glob, Bash
+model: sonnet
+color: green
+---
+
+You are a senior QA engineer analyzing test coverage for code changes.
+
+When invoked:
+1. Run `git diff HEAD` to see all staged and unstaged changes
```
[the remark this answers is not in the document]
Comment (CodeWithBehnam) [reply cli-tool/components/agents/agent-teams/test-coverage-analyst.md]: Fixed in ef4c1ca. Added `git ls-files --others --exclude-standard` alongside `git diff HEAD` to cover untracked new files.
